mock/mock.go

```diff
@@ -669,17 +669,14 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 	for i := 0; i < maxArgCount; i++ {
 		var actual, expected interface{}
 
-		if len(objects) <= i {
-			actual = "(Missing)"
-		} else {
-			actual = objects[i]
+		if len(objects) <= i || len(args) <= i {
```

Comment:
@dnathe4th We should ideally do this as part of `mock.On` function, and we need to consider `varargs`, and have to be tested extensively, lets create an issue and raise PR separately. I would suggest just with previous commit along with test case would be good to close the PR.

Reply:
I replied on #587 I am not sure it is possible to do this check as part of `mock.On` because there is no way for a function on `*mock.Mock` to be aware of what struct has embedded it as far as I could tell. 

Functionally these two should be the same, but if you like the previous commit better with the extra checks in the conditional at the end I can revert back to that. This version seems strictly better to me from a readability perspective since it has an explicit check on arg length.

From my digging through the existing tests and my understanding of the `reflect` package, it doesn't seem like varargs needs any special handling since tests like `Test_Mock_On_WithMixedVariadicFunc` pass on this PR, but maybe I am misunderstanding what is needed here.

Reply:
yep, checked varargs are passed as `slice`, so we're good.

lets just pull this out of loop, or print it only once when required.

Reply:
Even though this is inside the loop it should only print once because I return early. I implemented it like this do the `differences` counter would still return the total number of errors against the mock, which seemed more correct than returning before the loop with a return value of `differences=1`

If you feel strongly I can pull it outside the loop.

Reply:
`len(objects), len(args)` wouldn't change inside the loop, so wouldn't this still return `differences=1` in first loop itself.

Reply:
Hey @devdinu , where did we end up on this? Is this change only acceptable with the check before the loop?

Reply:
@dnathe4th The change looks good. @ernesto-jimenez could help in merging this.